gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/GradeStatisticsPanel.java

```diff
@@ -0,0 +1,135 @@
+package org.sakaiproject.gradebookng.tool.panels;
+
+import java.util.List;
+import java.util.Map;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.Collections;
+
+import org.apache.wicket.ajax.AjaxRequestTarget;
+import org.apache.wicket.ajax.markup.html.AjaxLink;
+import org.apache.wicket.extensions.ajax.markup.html.modal.ModalWindow;
+import org.apache.wicket.markup.html.basic.Label;
+import org.apache.wicket.markup.html.panel.Panel;
+import org.apache.wicket.model.IModel;
+import org.apache.wicket.model.Model;
+import org.apache.wicket.model.StringResourceModel;
+import org.apache.wicket.spring.injection.annot.SpringBean;
+import org.sakaiproject.gradebookng.business.GradebookNgBusinessService;
+import org.sakaiproject.gradebookng.business.model.GbGradeInfo;
+import org.sakaiproject.gradebookng.business.model.GbStudentGradeInfo;
+import org.sakaiproject.service.gradebook.shared.Assignment;
+
+public class GradeStatisticsPanel extends Panel {
+
+	private static final long serialVersionUID = 1L;
+
+	private final ModalWindow window;
+
+	@SpringBean(name = "org.sakaiproject.gradebookng.business.GradebookNgBusinessService")
+	protected GradebookNgBusinessService businessService;
+
+	public GradeStatisticsPanel(final String id, final IModel<Long> model, final ModalWindow window) {
+		super(id, model);
+		this.window = window;
+	}
+
+	@Override
+	public void onInitialize() {
+		super.onInitialize();
+
+		final Long assignmentId = ((Model<Long>) getDefaultModel()).getObject();
+
+		Assignment assignment = businessService.getAssignment(assignmentId.longValue());
+
+		add(new Label("title", new StringResourceModel("label.statistics.title",
+				null, new Object[] { assignment.getName() }).getString()));
+
+		List<GbStudentGradeInfo> gradeInfo = businessService.buildGradeMatrix(Arrays.asList(assignment));
```

Comment:
Thinking about this some more, this method is probably a bit heavy to call for this purpose as it does all of the category/course grade calculations, sorting, permissions etc. We could probably build a cut down method for this purpose at some point.